cli/lib/server/index.js

```diff
@@ -1,26 +1,73 @@
+import glob from "glob";
 import path from "path";
-import sirv from "sirv";
+import serveStatic from "serve-static";
 import { App } from "@tinyhttp/app";
-import { readFile } from "../helper/files.js";
-import { MANIFEST_FILENAME } from "../constants.js";
+import { json } from "milliparsec";
+import { cors } from "@tinyhttp/cors";
+import { readFile, writeFile } from "../helper/files.js";
+import { CONFIG_FILENAME, MANIFEST_FILENAME } from "../constants.js";
+import buildManifest from "../app/run/buildManifest.js";
 
-export const startServer = ({ outputFoldername }) => {
+function createPubSub() {
+  const pubSub = {
+    subscribers: [],
+    sub: function (subscriber) {
+      this.subscribers.push(subscriber);
+      return subscriber;
+    },
+    pub: function (change) {
+      this.subscribers.forEach((subscriber) => subscriber(change));
+    },
+    unsub: function (subscriber) {
+      this.subscribers = this.subscribers.filter((sub) => sub !== subscriber);
+    },
+  };
+  return pubSub;
+}
+
+export const startServer = (opts) => {
   const app = new App();
+  const pubSub = createPubSub();
 
   app
-    .use(function someMiddleware(req, res, next) {
-      res.setHeader("Access-Control-Allow-Origin", "*");
-      next();
-    })
-    .use("/assets", sirv(outputFoldername))
+    .use(json())
+    .use(cors({ origin: "*", allowedHeaders: "*" }))
+    .options("*", cors())
+    .use("/assets", serveStatic(opts.outputFoldername, { maxAge: 0 }))
     .get("/", (_, res) => {
       res.send("<h1>lode API is running</h1>");
     })
     .get("/manifest", (req, res) => {
-      const file = readFile(path.join(outputFoldername, MANIFEST_FILENAME));
+      const file = readFile(
+        path.join(opts.outputFoldername, MANIFEST_FILENAME)
+      );
       res.status(200).send(file);
     })
+    .get("/changes", (req, res) => {
+      const listener = (change) => {
+        const timestamp = new Date().getMilliseconds();
+        const manifest = readFile(
+          path.join(opts.outputFoldername, MANIFEST_FILENAME)
+        );
+        res
+          .status(200)
+          .send({ change, timestamp, manifest: JSON.parse(manifest) });
+        pubSub.unsub(listener);
```

Comment:
this means that messages being sent in between will not be noticed by the client.
ok for me, as we're pragmatic.